### PR TITLE
Prevent mentee from accessing mentee application after the state

### LINKED ISSUE #296: Mentee application shouldn't be accessible after mentee application state
**Describe the bug**
- Mentee application shouldn't be accessible after mentee application state

**To Reproduce**
After `MENTEE_APPLICATION` 
1. Go to `/program/{menteeId}/mentee/apply`
2. See the application UI

**Expected behavior**
- After `MENTEE_APPLICATION` state the mentee application UI shouldn't be accessible

**Screenshots**
image

**Desktop (please complete the following information):**
 - OS: MacOS Monteray
 - Browser: Chrome

### PR DESCRIPTION
**Pull request by anjula-sack**
## Purpose

- The purpose of this PR is to fix #296 

## Goals

- Prevent mentee from accessing mentee application after the `MENTEE_APPLICATION` state
## Approach

- Checked for the programme state and redirect the user 

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/scenes/Home/scenes/RequestMentors/scenes/Application/index.tsx [hunk 1]
```diff
@@ -28,8 +29,19 @@ function MenteeApplication() {
...
     getMentee();
+    getProgram();
   }, []);
 
+  const getProgram = async () => {
+    setIsLoading(true);
+    const program = await getProgramDetails(programId);
+    if (program && program.state !== 'MENTEE_APPLICATION') {
+      history.push(`/program/${programId}`);
+      history.go(0);
```
Comment (Gravewalker666) [inline src/scenes/Home/scenes/RequestMentors/scenes/Application/index.tsx]: What does this do?

Comment (anjula-sack) [reply src/scenes/Home/scenes/RequestMentors/scenes/Application/index.tsx]: reloads the page

Comment (Gravewalker666) [reply src/scenes/Home/scenes/RequestMentors/scenes/Application/index.tsx]: I don't understand don't we need to send them back to home??

Comment (anjula-sack) [reply src/scenes/Home/scenes/RequestMentors/scenes/Application/index.tsx]: oops wrong path yeah it's has to be home

Comment (Gravewalker666) [reply src/scenes/Home/scenes/RequestMentors/scenes/Application/index.tsx]: Did you test the app?

**Review by Gravewalker666 — APPROVED** [review]

**Pull request merged**